chrome/content/zotero/tabs.js

```diff
@@ -569,6 +570,18 @@ var Zotero_Tabs = new function () {
 		this.select(this._tabs[this._tabs.length - 1].id);
 	};
 
+	/**
+	 * Return focus into the reader of the selected tab.
+	 * Required to move focus from the tab into the reader after drag.
+	 */
+	this.refocusReader = function () {
+		var reader = Zotero.Reader.getByTabID(this._selectedID);
+		if (!reader) return;
+		setTimeout(() => {
+			reader.focus();
+		}, 100);
```

Comment:
We really want to avoid arbitrary delays. Can we watch for a focus event on the tab and focus the reader after that (if necessary with some maximum delay, if we think there are cases where the tab isn't focused)?

Reply:
> I assume that last time my VM was just being a bit glitchy, which is why I thought the 100 delay was required.

Generally it helps to troubleshoot this sort of thing with the environment somehow slowed down (by running something CPU intensive alongside), otherwise people with slower systems will start reporting something like that and we won't be able to reproduce.

Reading the code and thinking about it, it would make sense if running this in the next event loop may be enough (without the timeout), but it might help to make sure.

Reply:
Good call! I'll keep that in mind for the future. 

I ran [CpuStress](https://learn.microsoft.com/en-us/sysinternals/downloads/cpustres) at ~75% CPU usage and re-focusing of the reader still worked fine without any added delay to `setTimeout`.

I also booted up my old windows 10 laptop, and it worked there as well. So it does seem like the last time was a fluke of some kind.